index.js

```diff
@@ -1,6 +1,85 @@
 /* jshint node: true */
 'use strict';
 
+var path = require('path');
+var Funnel = require('broccoli-funnel');
+var replace = require('broccoli-string-replace');
+var mergeTrees = require('broccoli-merge-trees');
+var VersionChecker = require('ember-cli-version-checker');
+
 module.exports = {
-  name: 'ember-a11y'
+  name: 'ember-a11y',
+
+  init: function() {
+    if (this._super.init) {
+      this._super.init.apply(this, arguments);
+    }
+
+    var versionChecker = new VersionChecker(this);
+    versionChecker.for('ember-cli', 'npm').assertAbove('0.2.0');
+
+    var emberVersion = versionChecker.for('ember', 'bower');
+    var isGlimmer = (emberVersion.gt('2.9.0-beta') && emberVersion.lt('2.9.0')) || emberVersion.gt('2.10.0-alpha');
+    var isHTMLBars = emberVersion.gt('1.10.0-beta') && !isGlimmer;
+
+    if (isGlimmer) {
+      this.versionSpecificPath = path.join(this.root, 'glimmer-version');
+      this.isGlimmer = isGlimmer;
+      // this.options.addons.blacklist.push('ember-getowner-polyfill');
+    } else if (isHTMLBars) {
+      this.isHTMLBars = isHTMLBars;
+      this.versionSpecificPath = path.join(this.root, 'htmlbars-version');
+    }
+  },
+
+  treeForAddon: function() {
+    var trees = [this._super.treeForAddon.apply(this, arguments)];
+
+    // include `ember-internals` module ONLY for htmlbars
+    if (this.isHTMLBars) {
+      trees.push(new Funnel(this.versionSpecificPath, {
+        files: ['ember-internals.js'],
+        destDir: 'modules/' + this.name
+      }));
+    }
+
+    return mergeTrees(trees, { overwrite: true });
+  },
+
+  treeForTemplates: function() {
```

Comment:
I'm glad you did all of the work to get just the correct contents into each tree.